engine/access/rpc/backend/backend_execution_results.go

```diff
@@ -10,10 +10,18 @@ import (
 
 type backendExecutionResults struct {
 	executionResults storage.ExecutionResults
+	seals            storage.Seals
 }
 
 func (b *backendExecutionResults) GetExecutionResultForBlockID(ctx context.Context, blockID flow.Identifier) (*flow.ExecutionResult, error) {
-	result, err := b.executionResults.ByBlockID(blockID)
```

Comment:
Since we have removed the index, now we query sealed result by getting seal by blockID first, then getting result by seal.ResultID